keylime/httpclient_requests.py

```diff
@@ -1,53 +1,39 @@
 import http.client
 
-def request(method, url, port, params=None, data=None,context=None):
-    """
-    http client connection handler
+
+def request(method, host, port, url=None, data=None, context=None,
```

Comment:
Thanks for the pull request @kaifeng !

I think you might have meant to change `url=None` to `params=None` (this is why CI failed).

Reply:
Thanks @lukehinds , Ideally I think it's good to rename the params to url, but it appears the method are reference in many places, so will make a change at a later time.

Reply:
I do agree, its currently not really params and more paths. 

The handler does need to rework. I am introducing an Authorization Bearer so we will need to be able to handle headers. So the whole logic of conditionals checking for `data` , ssl.context and headers needs to be re-thought. I might even move us over to using requests instead of http.client.

Thanks for your contribution

Reply:
Agreed, the requests is a very good candidate, which is also widely used in the openstack, we can iterate though.